### PR TITLE
Implement some packets

### PR DESCRIPTION
**Pull request by SirGhostal**

### CODE REVIEW

File: minecraft/networking/packets/clientbound/play/__init__.py [hunk 1]
```diff
@@ -72,7 +74,22 @@ def get_id(context):
...
+        {'reduced_debug_info': Boolean}
+    ])
+
+
+class ServerDifficultyPacket(Packet):
+    @staticmethod
+    def get_id(context):
+        return 0x0D if context.protocol_version >= 332 else \
+               0x0E if context.protocol_version >= 318 else \
+               0x0D if context.protocol_version >= 67 else \
```
Comment (TheSnoozer) [inline minecraft/networking/packets/clientbound/play/__init__.py]: I guess for the old pre-release versions it wouldn't make a huge difference, but the packet id changed in [15w36d ( protocol 70) to 0x0D](https://wiki.vg/index.php?title=Pre-release_protocol&oldid=6925).
aka.
```python
0x0D if context.protocol_version >= 70 else \
```

File: minecraft/networking/packets/clientbound/play/__init__.py [hunk 2]
```diff
@@ -184,6 +201,27 @@ def get_id(context):
...
+               0x33 if context.protocol_version >= 70 else \
+               0x07
+
+    packet_name = 'respawn'
+    get_definition = staticmethod(lambda context: [
+        {'dimension': Integer},
+        {'difficulty': UnsignedByte},
+        {'game_mode': UnsignedByte},
+        {'level_type': String}
+    ])
```
Comment (TheSnoozer) [inline minecraft/networking/packets/clientbound/play/__init__.py]: What you think to add a convenience subclass here that denotes the dimension/difficulty/game_mode?
The information could be shared with the [JoinGamePacket](https://github.com/ammaraskar/pyCraft/blob/master/minecraft/networking/packets/clientbound/play/__init__.py#L57)

```python
    class Difficulty(Enum):
        PEACEFUL = 0
        EASY = 1
        NORMAL = 2
        HARD = 3

    class Dimension(Enum):
        NETHER = -1
        OVERWORLD = 0
        END = 1

    class Gamemode(Enum):
        SURVIVAL = 0
        CREATIVE = 1
        ADVENTURE = 2
        SPECTATOR = 3
```

File: minecraft/networking/packets/clientbound/play/__init__.py (hunk 1 above)
Comment (Zachy24) [reply minecraft/networking/packets/clientbound/play/__init__.py]: Good spot here. Looks like this is missing on the wiki page. Should have looked at https://joodicator.github.io/mc-dev-data/ instead ;)

Ill update the PR.

File: minecraft/networking/packets/clientbound/play/__init__.py (hunk 2 above)
Comment (Zachy24) [reply minecraft/networking/packets/clientbound/play/__init__.py]: I like this idea. Making them available as a type  in https://github.com/ammaraskar/pyCraft/blob/master/minecraft/networking/types/enum.py
would save us having to copy / paste the Enum class  in each packet definition...

**Comment by joodicator** [comment]
Thank you for this pull request. This was very well written. I apologise for the incredible length of time it took for it to be merged.

**Pull request merged**